src/shared_modules/indexer_connector/src/indexerConnector.cpp

```diff
@@ -579,6 +580,67 @@ IndexerConnector::IndexerConnector(
         });
 }
 
+IndexerConnector::IndexerConnector(
+    const nlohmann::json& config,
+    const std::function<void(
+        const int, const std::string&, const std::string&, const int, const std::string&, const std::string&, va_list)>&
+        logFunction)
+{
+    if (logFunction)
+    {
+        Log::assignLogFunction(logFunction);
+    }
+
+    // Get index name.
+    m_indexName = config.at("name").get_ref<const std::string&>();
+
+    if (Utils::haveUpperCaseCharacters(m_indexName))
+    {
+        throw std::runtime_error("Index name must be lowercase.");
+    }
+
+    m_db = std::make_unique<Utils::RocksDBWrapper>(std::string(DATABASE_BASE_PATH) + "db/" + m_indexName);
```

Comment:
Repeated code between the two constructors.

Reply:
I'll create a common private method with the shared code.